src/corecel/cont/EnumBitset.hh

```diff
@@ -237,28 +240,37 @@ class Bitset<N>::reference
     CELER_CONSTEXPR_FUNCTION
     operator bool() const noexcept
     {
-        return (*word_pointer_ & Bitset::mask(bit_pos_)) != 0;
+        return (*word_pointer_ & this->mask()) != 0;
     }
 
     //! To support b[i].flip();
     CELER_CONSTEXPR_FUNCTION
     reference& flip() noexcept
     {
-        *word_pointer_ ^= Bitset::mask(bit_pos_);
+        *word_pointer_ ^= this->mask();
         return *this;
     }
 
   private:
     word_type* word_pointer_{nullptr};
     size_type bit_pos_{0};
+
+    CELER_CONSTEXPR_FUNCTION word_type mask() const noexcept
+    {
+        return word_type(1) << bit_pos_;
```

Comment:
Performance-wise, this is the crux of this functionality (setting might be rare but testing might not be).
Thus there is a shift operation plus a binary or operation for each test.
An alternative implementation is to use (not as easily enforceable from a C++ stand point )
```
enum class MscModelSelection
{
    none,
    urban = 1, 
    wentzelvi = 2, /* really 1  << 1 */
    somethingelse = 4, /* really 1 << 2 */
    size_
};
```
The `Bitset` associated with that now wouldn't have to do the shift operation at run-time (as long as there is less than 64 enum values).

Reply:
@pcanal Being constexpr, this should be optimized away. Since we define `CELER_CONSTEXPR_FUNCTION` as `constexpr inline __attribute__((always_inline))` it even gets replaced by a scalar value in completely "unoptimized" code:
https://godbolt.org/z/eYehqebqa

So there's no penalty to indexing with `i` versus `2^i`.

Reply:
`constexpr` is not a guarantee, it is a possibility that depends on all inputs to also be `constexpr`.  Unlike the example you used, `bit_pos_` is a data member and thus not guaranteed to be `constexpr`.  `bit_pos_` is in turn set from the parameter to `operator[]`  and so `options_.msc[MMS::urban]` would indeed have the possibility of executing the shift at compile time but `MscModelSelection m = ...; .... then somewhere else/indirect .... options_.msc[m]` would not be able to.   If only the former  is intended to be used, we could/should/might add code to make the later fail to compile.

Reply:
Actually I have a potentially easy fix for at least part of your problem... just store `mask_` instead of `bit_pos_`.

Reply:
> since I can't think of any case where we want more than 32 separate flags...

I agree that a solution that support 32 flags (or 64 with a `uint64_t`) will be sufficient.

Reply:
> what's 1 line with -O1 

For `constexpr` inputs or run-time inputs?

Reply:
Storing `mask_` saved like one instruction in `-O0`, looking closer at the assembly the main overhead is stack pointer manipulation 🤷 

This is for a constexpr enum access but run-time value being accessed: `get_b` above compiles down to `get_manual`.

Maybe we've approached this wrong: perhaps we just need an "enum bitset" with a single word... @esseivaju what circumstance would require a bitset with compile-time length > 32?

Reply:
I can't foresee many situations where a length larger than 32bits would be needed. We briefly touched on that in the PR implementing the bitset, since `N` is a constexpr, `N=1` reduces to a single word. So, I'm not sure what would be the benefit of supporting a single word only as it doesn't add that much complexity.

Reply:
> This is for a constexpr enum access but run-time value being accessed:

The part that was surprising was `what's 1 line with -O1`.   What is the one instructions that does both shift and test?  (More likely I miss assumed what that was covering :) ).